src/main/java/com/gregtechceu/gtceu/utils/GTUtil.java

```diff
@@ -485,6 +486,34 @@ public static ItemStack loadItemStack(CompoundTag compoundTag) {
         }
     }
 
+    public static final String[] FIX_TAGS = {
+            "currentMilliBucketsPerTick",
+            "globalTransferSizeMillibuckets",
+            "minValue",
+            "maxValue",
+    };
+
+    // This is necessary for updating from old versions due to FluidStack long -> int changes
+    // Any fluid-related long tags need to be turned into int tags
+    public static void fixFluidTags(CompoundTag tag) {
+        if (tag.contains("cover", Tag.TAG_COMPOUND)) {
+            CompoundTag t = tag.getCompound("cover");
+            for (String key : t.getAllKeys()) {
+                var cover = t.getCompound(key);
+                var id = cover.getCompound("uid").getString("id");
+                if ((id.toLowerCase().contains("fluid") || id.toLowerCase().contains("pump"))) {
+                    var data = cover.getCompound("payload").getCompound("d");
+                    for (String fix_key : FIX_TAGS) {
+                        if (data.contains(fix_key, Tag.TAG_LONG)) {
+                            var l = data.getLong(fix_key);
+                            data.putInt(fix_key, Ints.saturatedCast(l));
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
please put this in, idk, `common/datafixer`. that's the path where the actual datafixers will go.

Reply:
Moved to `common/datafixer/TagFixer`